otter/integration/lib/autoscale.py

```diff
@@ -106,52 +109,29 @@ def wait_for_N_servers(
             Otherwise, an exception is raised.
         """
 
-        clock = clock or reactor
-
-        class Looper(object):
-            """Inner-class to support looping until either of two conditions
-            holds: (1) we see at least `servers_desired` servers in existence,
-            or (2) a timeout is reached.
-
-            :param ScalingGroup scaling_group: The scaling group that this
-                looper should poll the state of.
-            """
-
-            def __init__(self, scaling_group):
-                self.elapsed_time = 0
-                self.scaling_group = scaling_group
-                # To be filled in later.
-                self.loopingCall = None
-
-            def loop(self):
-                if self.elapsed_time < timeout:
-                    self.elapsed_time += period
-                    return self.check_servers()
-                else:
-                    raise TimeoutError(
-                        "Spent %ds, polling every %ds, timeout." % (
-                            self.elapsed_time, period
-                        )
-                    )
-
-            def ensure_200_result(self, state_results):
-                if state_results[0] != 200:
-                    raise Exception("Scaling group not found.")
-                active_servers = state_results[1]["group"]["active"]
-                if len(active_servers) >= servers_desired:
-                    self.loopingCall.stop()
-
-            def check_servers(self):
-                d = self.scaling_group.get_scaling_group_state(rcs)\
-                    .addCallback(self.ensure_200_result)
-                return d
-
-        looper = Looper(self)
-        lc = LoopingCall(looper.loop)
-        lc.clock = clock
-        looper.loopingCall = lc
-        d = lc.start(period).addCallback(lambda _: rcs)
-        return d
+        def check(state):
+            if state[0] != 200:
+                raise BreakLoopException("Scaling group not found.")
+            servers_active = len(state[1]["group"]["active"])
+            if servers_active == servers_desired:
+                return rcs
+
+            raise TransientRetryError()
+
+        def poll():
+            return self.get_scaling_group_state(rcs).addCallback(check)
+
+        return retry_and_timeout(
+            poll, timeout,
+            can_retry=transient_errors_except(BreakLoopException),
+            next_interval=repeating_interval(period),
+            clock=clock or reactor,
+            deferred_description=(
+                "Server count {} not reached in {}s timeout.".format(
```

Comment:
Similarly, since [TimedOutException](https://github.com/rackerlabs/otter/blob/master/otter/util/deferredutils.py#L56) already formats it with the correct timeout time, maybe `"Waiting for server count to reach {0} state".format(servers_desired)`?